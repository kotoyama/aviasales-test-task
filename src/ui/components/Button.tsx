import React from 'react'
import { styled } from '@linaria/react'

type Props = React.ButtonHTMLAttributes<HTMLButtonElement>

export const Button: React.FC<Props> = ({ children, ...props }) => (
  <StyledButton type="button" {...props}>
    {children}
  </StyledButton>
)

const StyledButton = styled.button`
  width: 100%;
  height: 50px;
  font: inherit;
  letter-spacing: 0.5px;
  text-align: center;
  text-transform: uppercase;
  background-color: var(--color-white);
  border: 1px solid var(--color-lightgray);

  &:focus {
    box-shadow: 0 0 0 2px var(--color-button-focus);
  }
`
