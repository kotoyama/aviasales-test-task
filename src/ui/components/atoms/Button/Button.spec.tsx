import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Button } from '../Button'

describe('Button component', () => {
  const buttonText = 'Click me'
  const clickHandler = jest.fn()

  test('should render correctly', () => {
    const component = renderer.create(
      <Button onClick={clickHandler}>{buttonText}</Button>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should be clickable', () => {
    const { getByText } = render(
      <Button onClick={clickHandler}>{buttonText}</Button>,
    )
    userEvent.click(getByText(buttonText))
    expect(clickHandler).toHaveBeenCalledTimes(1)
  })
})
