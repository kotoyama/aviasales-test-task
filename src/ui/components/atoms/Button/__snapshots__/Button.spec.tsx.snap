// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly 1`] = `
<button
  className="sfxbk49"
  onClick={[MockFunction]}
  type="button"
>
  Click me
</button>
`;
